test/src/SpendPermissions/permit.t.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {Test, console2} from "forge-std/Test.sol";
+
+import {SpendPermissions} from "../../../src/SpendPermissions.sol";
+
+import {SpendPermissionsBase} from "../../base/SpendPermissionsBase.sol";
+
+contract PermitTest is Test, SpendPermissionsBase {
+    function setUp() public {
+        _initializeSpendPermissions();
+    }
+
+    function test_permit_revert_unauthorizedRecurringAllowance() public {}
+    function test_permit_success() public {}
```

Comment:
same thins as `approve` for when there are desired effects, we want tests to make sure those effects happen
- `test_approve_success_isAuthorized`
- `test_approve_success_emitsEvent`